tests/Endpoints/SearchTest.php

```diff
@@ -139,6 +139,112 @@ public function testExceptionIfNoIndexWhenSearching(): void
         $index->search('prince');
     }
 
+    public function testParametersCropMarker(): void
+    {
+        $response = $this->index->updateFilterableAttributes(['title']);
+        $this->index->waitForTask($response['uid']);
+
+        $response = $this->index->search('blood', [
+            'limit' => 1,
+            'attributesToCrop' => ['title'],
+            'cropLength' => 2,
+        ]);
+
+        $this->assertArrayHasKey('_formatted', $response->getHit(0));
+        $this->assertSame('…Half-Blood…', $response->getHit(0)['_formatted']['title']);
+
+        $response = $this->index->search('blood', [
+            'limit' => 1,
+            'attributesToCrop' => ['title'],
+            'cropLength' => 2,
+        ], [
+            'raw' => true,
+        ]);
+
+        $this->assertArrayHasKey('_formatted', $response['hits'][0]);
+        $this->assertSame('…Half-Blood…', $response['hits'][0]['_formatted']['title']);
```

Comment:
Should we do two tests in one test function?

Reply:
The first line checks if the key `_formatted` exists before testing the value contained in it.

Reply:
> Should we do two tests in one test function?

It depends, unit tests usually have one assertion, but integration/e2e tests have more assertions because of the computational cost of their setup. In other words, if we wrote assertions for e2e as we do for unit testing, the suite will take forever to finish!